lib/tree.rb

```diff
@@ -16,40 +16,137 @@ def initialize
     @root = nil
   end
 
-  # Time Complexity: 
-  # Space Complexity: 
+  # Time Complexity: o(log n) if it's balanced; worst case is o(n)  
+  # Space Complexity: o(n)
   def add(key, value)
```

Comment:
👍  Space & Time complexities are both O(log n) unless the tree is unbalanced, in which case they are O(n).